File: Leetcode/122/solution.java
// https://leetcode.com/problems/best-time-to-buy-and-sell-stock-ii/description/
// O(n) runtime, O(1) space
class Solution {
    public int maxProfit(int[] prices) {
        int maxprofit = 0;
        for (int i = 1; i < prices.length; i++) {
            if (prices[i] > prices[i - 1])
                maxprofit += prices[i] - prices[i - 1];
        }
        return maxprofit;
    }
}

File: Leetcode/122/solution.py
# https://leetcode.com/problems/best-time-to-buy-and-sell-stock-ii/description/
# O(n) runtime, O(1) space
class Solution(object):
    def maxProfit(self, prices):
        # Sweet python one-liner
        return sum(max(prices[i + 1] - prices[i], 0) for i in range(len(prices) - 1))
